### PR TITLE
Add type hints - and add tests for themes.py

### LINKED ISSUE #284: Add type hints to all the Python scripts
This article explains the benefits of type hints very nicely:

- [Why I started using Python type annotations – and why you should too](https://florimond.dev/en/posts/2018/07/why-i-started-using-python-type-annotations-and-why-you-should-too/)

This issue is mainly to save notes on the mechanics of how I'm doing it, to help future maintainers.

### PR DESCRIPTION
**Pull request by claremacrae**
## Type Hints

- Added type hints to all Python code
- The Python GitHub Action now checks type hints (after running the tests)

## Convenience Scripts

- Added `check_type_hints.sh` script to use `mypy` to check that hints are present and correct, and that arguments to functions are correct
- Added `add_type_hints.sh` script that uses `monkeytype` to run code and add missing type hints

## Automated Tests

- Added `test_themes.py` that exercises a lot of the code in `themes.py`
- Added a directory `.github/scripts/tests/sample_data/themes/` containing Minimal (greatly reduced in size) and Christmas themes, so these can be created easily for writing tests
- Added helper function `get_saved_sample_data_for_theme()` to read those sample theme files

## Other Info

This fixes #284 - and there are some notes in that issue with some tips for if stuck on getting type hints to work (that will eventually be moved to a note in the Hub).

### CODE REVIEW

File: .github/scripts/requirements.txt [hunk 1]
```diff
@@ -3,3 +3,7 @@ pyyaml
 requests
 approvaltests==3.5.0
 pytest
+mypy
+types-PyYAML
+types-requests
+monkeytype
```
Comment (argenos) [inline .github/scripts/requirements.txt:9]: Can you please mention why we need monkeytype?

**Review by argenos — APPROVED** [review]

File: .github/scripts/requirements.txt (hunk 1 above)
Comment (claremacrae) [reply .github/scripts/requirements.txt:9]: @argenos Sure:

> Added `add_type_hints.sh` script that uses monkeytype to run code and add missing type hints

It means that if someone adds some new code and doesn't have an IDE that adds types easily, they can run the code and monkeytype will add a starting point for type hints.

Comment (argenos) [reply .github/scripts/requirements.txt:9]: Is it an optional dependency in that case? Adding it to requirements.txt will install it regardless of whether one uses an IDE or prefers to add the types manually (if I understood right what it does).

Comment (claremacrae) [reply .github/scripts/requirements.txt:9]: OK sure. I'll comment out the monkeytype dependency, and add comments in the requirements.txt to divide the groups up and explain them

**Pull request merged**